Flaky `test_async_task_group_call_later_executes_delayed_task_in_background`

<img width="1112" alt="Screen Shot 2022-07-19 at 10 49 35 AM" src="https://user-images.githubusercontent.com/3309802/179780288-450fd656-b0b5-4cf3-9d95-aecb9a29db82.png">

Seems to have started failing on July 4, only on macs.

```
    async def test_async_task_group_call_later_executes_delayed_task_in_background():
        group = AsyncTaskGroup()
        flag = False
    
        async def set_flag():
            nonlocal flag
            flag = True
    
        start = timemod.monotonic()
        assert group.call_later(1, set_flag) is None
        assert len(group) == 1
        # the task must complete in exactly 1 event loop cycle
        await asyncio.sleep(1)
        await _wait_for_n_loop_cycles(2)
        end = timemod.monotonic()
>       assert len(group) == 0
E       assert 1 == 0
E        +  where 1 = len(<distributed.core.AsyncTaskGroup object at 0x13aa5d630>)

distributed/tests/test_core.py:128: AssertionError
```

https://github.com/dask/distributed/runs/7358684710?check_suite_focus=true#step:11:1916

cc @hendrikmakait @graingert

--- a/distributed/tests/test_core.py
+++ b/distributed/tests/test_core.py
@@ -114,11 +114,8 @@ async def test_async_task_group_call_later_executes_delayed_task_in_background()
     group = AsyncTaskGroup()
     ev = asyncio.Event()
 
-    async def set_event():
-        ev.set()
-
     start = timemod.monotonic()
-    assert group.call_later(1, set_event) is None
+    assert group.call_later(1, ev.set) is None
     assert len(group) == 1
     await ev.wait()
     end = timemod.monotonic()

--- a/distributed/tests/test_core.py
+++ b/distributed/tests/test_core.py
@@ -112,21 +112,19 @@ async def test_async_task_group_call_soon_executes_task_in_background():
 @gen_test()
 async def test_async_task_group_call_later_executes_delayed_task_in_background():
     group = AsyncTaskGroup()
-    flag = False
+    ev = asyncio.Event()
 
-    async def set_flag():
-        nonlocal flag
-        flag = True
+    async def set_event():
+        ev.set()
 
     start = timemod.monotonic()
-    assert group.call_later(1, set_flag) is None
+    assert group.call_later(1, set_event) is None
     assert len(group) == 1
-    # the task must complete in exactly 1 event loop cycle
-    await asyncio.sleep(1)
-    await _wait_for_n_loop_cycles(2)
+    await ev.wait()
     end = timemod.monotonic()
+    # the task must be removed in exactly 1 event loop cycle
+    await _wait_for_n_loop_cycles(2)
     assert len(group) == 0
-    assert flag
     assert end - start > 1 - timemod.get_clock_info("monotonic").resolution
 
 

--- a/distributed/tests/test_core.py
+++ b/distributed/tests/test_core.py
@@ -120,16 +120,12 @@ async def test_async_task_group_call_later_executes_delayed_task_in_background()
     start = timemod.monotonic()
     assert group.call_later(1, set_event) is None
     assert len(group) == 1
-
     await ev.wait()
     end = timemod.monotonic()
     # the task must be removed in exactly 1 event loop cycle
     await _wait_for_n_loop_cycles(2)
     assert len(group) == 0
-    duration = end - start
-    clock_resolution = timemod.get_clock_info("monotonic").resolution
-    assert duration > 1 - clock_resolution
-    assert duration < 1.01 + clock_resolution
+    assert end - start > 1 - timemod.get_clock_info("monotonic").resolution
 
 
 def test_async_task_group_close_closes():

--- a/distributed/tests/test_core.py
+++ b/distributed/tests/test_core.py
@@ -120,12 +120,16 @@ async def test_async_task_group_call_later_executes_delayed_task_in_background()
     start = timemod.monotonic()
     assert group.call_later(1, set_event) is None
     assert len(group) == 1
+
     await ev.wait()
     end = timemod.monotonic()
     # the task must be removed in exactly 1 event loop cycle
     await _wait_for_n_loop_cycles(2)
     assert len(group) == 0
-    assert end - start > 1 - timemod.get_clock_info("monotonic").resolution
+    duration = end - start
+    clock_resolution = timemod.get_clock_info("monotonic").resolution
+    assert duration > 1 - clock_resolution
+    assert duration < 1.01 + clock_resolution
 
 
 def test_async_task_group_close_closes():

